python/lsst/meas/algorithms/matcherSourceSelector.py

```diff
@@ -162,4 +162,71 @@ def _isUsable(self, source):
             and not source.get(self.fluxFlagKey) \
             and self._goodSN(source)
 
+
+class MatcherPessimisticSourceSelectorTask(MatcherSourceSelectorTask):
+    """
+    !Select sources that are useful for matching.
+
+    Good matching sources have high signal/noise, are non-blended. They need not
+    be PSF sources, just have reliable centroids.
+    """
+    def _getSchemaKeys(self, schema):
+        """Extract and save the necessary keys from schema with asKey."""
+        self.parentKey = schema["parent"].asKey()
+        self.centroidXKey = schema["slot_Centroid_x"].asKey()
+        self.centroidYKey = schema["slot_Centroid_y"].asKey()
+        self.centroidFlagKey = schema["slot_Centroid_flag"].asKey()
+
+        fluxPrefix = "slot_%sFlux_" % (self.config.sourceFluxType,)
+        self.fluxField = fluxPrefix + "flux"
+        self.fluxKey = schema[fluxPrefix + "flux"].asKey()
+        self.fluxFlagKey = schema[fluxPrefix + "flag"].asKey()
+        self.fluxSigmaKey = schema[fluxPrefix + "fluxSigma"].asKey()
+
+        self.edgeKey = schema["base_PixelFlags_flag_edge"].asKey()
+        self.interpolatedCenterKey = schema["base_PixelFlags_flag_interpolatedCenter"].asKey()
+        self.saturatedKey = schema["base_PixelFlags_flag_saturated"].asKey()
```

Comment:
I strongly suggest you rewrite this to call the parent class version and then add the extra stuff, in order to simplify the code and reduce code duplication. E.g.:
```
 def _getSchemaKeys(self, schema):
    MatcherSourceSelectorTask._getSchemaKeys(self, schema)
    self.edgeKey = schema["base_PixelFlags_flag_edge"].asKey()
    ...
```
Please also use this technique for `_isUsable_vector` and try it for `_isUsable` (it might noticeably slow performance, but it might not).